mlrun/utils/notifications/notification_pusher.py

```diff
@@ -118,21 +118,43 @@ def __init__(
         ] = []
 
         for run in self._runs:
+            self._process_run(run)
+
+    def _process_run(self, run):
+        try:
             if isinstance(run, dict):
                 run = mlrun.model.RunObject.from_dict(run)
 
             for notification in run.spec.notifications:
-                try:
-                    notification.status = run.status.notifications.get(
-                        notification.name
-                    ).get("status", mlrun.common.schemas.NotificationStatus.PENDING)
-                except (AttributeError, KeyError):
-                    notification.status = (
-                        mlrun.common.schemas.NotificationStatus.PENDING
-                    )
+                self._process_notification(notification, run)
+        except Exception as exc:
+            logger.warning(
+                "Failed to load notification from run",
+                run_uid=run.metadata.uid,
+                error=mlrun.errors.err_to_str(exc),
+            )
 
-                if self._should_notify(run, notification):
-                    self._load_notification(run, notification)
+    def _process_notification(self, notification, run):
+        try:
```

Comment:
remove the outer scope of `try` and move it to whoever call `_process_notification` and on the exception message, replace the `Failed to load notification` with `Failed to process notification`
this way you gain a correct warning log message and less indented code